### PR TITLE
saved recipes post functionality built

### PR DESCRIPTION
**Pull request by amysiu1028**
Saved recipes and post functionality built where only unique saved recipes are displayed onto the saved recipe page.

Next steps:

Frontend:
- fix typescript errors
- message project manager about cypress or if someone can start on that if styling is okay at the moment?
- last: redeployment on vercel 

BackEnd doing styling + FrontEnd:
- meet and preview what we have left to do so far
- update on where we are at on styling, add more styling
- Edit ShowRecipePage, and SingleFilteredRecipePage. I'm not sure if the database was updated for instructions to be an array of strings? If that's updated I can work on that and have it display like the ingredients.
- talk about a delete for part 2? (because right now we can do a post but we can't delete, or unsave the saved recipe? - Jen do you happen to know about this? Maybe walk through the app and see if that's possible? I am going to take a bit of a brain break. 

Backend: 
- 500 Network error: there is a initial get network error (image below) that keeps popping up for 2 letter of the searchTerm... is this a backend issue? I am not focusing too much on it because we are getting the data when we do the search. But is this something that backend can take a look? It happens with po for pork and then chi for chicken

### CODE REVIEW

**Comment by amysiu1028** [comment]
Screen Shot 2024-02-07 at 7 22 38 PM
40cc-bb72-a1d3fb1a0c88">

**Comment by amysiu1028** [comment]
Screen Shot 2024-02-07 at 7 22 38 PM

**Comment by skim1027** [comment]
regarding the 500 errors, I am not sure how the request is happening, and wondering if there is a way we can make this fetch request only when the button is clicked? Or, I am also wondering if this error is doing anything when the button hasn't been pushed? Is it actually making a request every time a letter is entered? with our controller, if there is no recipes with the search term, we are supposed to be getting 404 error status.

As far as styling, I tried updating the log in page, which Jen is now taken over and updating other styling as well. It's under branch log_in_styling

I am going to update some styling for the show recipe page.

There is a PR for backend, waiting for Abdul's feedback, but that will implement the string to array. 

I think the only thing we really have left (if we are getting post and get request correctly) is styling?

File: src/Components/SavedRecipeCard/SavedRecipeCard.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import './SavedRecipeCard.css'
+import { RecipesData, smallRecipePageProps } from '../../types';
+import CookingPot from '../../Images /Cooking-pot.jpg'
+import { useNavigate } from 'react-router-dom';
+
+const SavedRecipeCard = ({ recipes, name, instructions, image_url, ingredients, updateSeletedFilteredRecipe }) => {
+  console.log("recipes outside function",recipes)
+
+//for now, we aren't going to navigate from the savedRecipe card to the more details page (I'd have to refactor the routing for this... and becuase we have one for webscrape and one from filteredpage to singleFilteredPage, I can't figure it out now. But maybe the routes will have to match somehow with all of them which I think is possible in Part 2)
```
Comment (Jnguyen615) [inline src/Components/SavedRecipeCard/SavedRecipeCard.tsx:9]: Lets talk about this tomorrow

File: src/Components/SavedRecipePage/SavedRecipesPage.tsx [hunk 2]
```diff
@@ -1,49 +1,122 @@
...
+        updateSavedRecipes(userSpecificRecipes);
+      });
+  }, [id]);
+
+  console.log('savedRecipes', savedRecipes);
+
+  // const uniqueRecipeFlat = savedRecipes.flat()
+  // console.log("uniqueRecipeFlat",uniqueRecipeFlat)
+
+  const uniqueRecipeNamesSet = new Set(savedRecipes.flat().map((recipe) => recipe.name));
```
Comment (Jnguyen615) [inline src/Components/SavedRecipePage/SavedRecipesPage.tsx:31]: I'm going to have to have you break this down for me, I only understand it a little

**Review by Jnguyen615 — APPROVED** [review]
There's some things I'll need some clarification on and other stuff we can talk through and hopefully find a solution for. So happy its working!

**Pull request merged**